cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts

```diff
@@ -366,8 +366,216 @@ describe("Device properties widget", function () {
       cy.cleanup();
       cy.deleteAllDevices();
     });
   });
-});
+
+  context('Last Measurements', function() {
+    const returnTypeFieldSelector = '#returnTypeField';
+    const expandedDataPointLabel = '.expanded .data-point-label';
+    const expandedInputNameLabel = '.expanded input[name="label"]';
+    const datapoint_selection_list = 'c8y-datapoint-selection-list';
+    const last_measurement_str = 'Last measurement';
+    const last_recent_measurement_str = 'Last measurement most recent'
+    const flowDataPointLabel = 's7aFlow → F';
+    const temperatureDataPointLabel = 's7aTemp → T';
+    const flowLatestDataPointLabel = 's7aFlowLatest → Cubic';
+    const assetPropertySelectorLabel = '[data-cy=asset-property-selector-label]';
+    const addDataPointsButton = '[title="Add data points"] ';
+    const computedPropertyConfigSaveButton = '[data-cy="computed-property-config-save-button"]';
+    const selector_button_computed_property_config = '[data-cy="asset-property-selector-config-computed-property-button"]';
+    const addDataPointConfigButton = '.card-footer > .btn';
+    const max_active_data_points_message = 'At maximum 1 active data points are allowed to be selected.';
+    const min_active_data_points_message= 'At least 1 active data points must be selected.';
+    before(function () {
+      cy.login();
+      function createMeasurement(deviceName, fragment, series, unit, value) {
+        const requestBody = {
+            deviceName: deviceName,
+            fragment: fragment,
+            series: series,
+            unit: unit,
+            value: value
+        };
+        cy.createMeasurement(requestBody);
+      }
+    
+      cy.apiCreateSimpleAsset([assetObject]);
+      for (let i = 1; i < 3; i++) {
+        cy.createDevice(`Device${i}`);
+        cy.apiAssignDevice([`Device${i}`],asset);
+        createMeasurement(`Device${i}`, 's7aFlow', 'F', 'KL', 45);
+        createMeasurement(`Device${i}`, 's7aTemp', 'T', 'K', 450);
+      }
+    });
+
+    beforeEach(function () {
+      cy.login();
+      cy.visitAndWaitUntilPageLoad(url);
+      cy.get(asset_properties_widget_elements.addWidgetButton).click();
+      cy.get(asset_properties_widget_elements.cardElement).eq(0).click();
+      cy.clickOnAsset(asset);
+      cy.get(`p[title='${device}']`).should('contains.text', device);
+      cy.chooseAssetOrDevice(device);
+    });
+
+    // Function to add a data point configuration
+    const addDataPointConfiguration = () => {
+      cy.get(asset_properties_widget_elements.addPropertyButton).click();
+      cy.selectProperty(last_measurement_str);
+      cy.get(asset_properties_widget_elements.selectButton).click();
+      cy.get('.card-title').should('have.text', 'Data points');
+      cy.intercept({ method: 'GET', url: '/inventory/managedObjects?**)' }).as(
+        'assetselectionresponse'
+      );
+      cy.get(addDataPointConfigButton).click();
+      cy.wait('@assetselectionresponse')
```

Comment:
Since these lines are used in many places, you can write a function for this.

Reply:
incorporated.